src/os/filestore/FileStore.cc

```diff
@@ -1873,10 +1898,17 @@ void FileStore::_do_op(OpSequencer *osr, ThreadPool::TPHandle &handle)
 
   osr->apply_lock.Lock();
   Op *o = osr->peek_queue();
-  apply_manager.op_apply_start(o->op);
+
+  if (!((g_conf->filestore_do_fast_sync) && (m_filestore_journal_writeahead))) {
```

Comment:
Or not const and set them in mount where you can return -EINVAL if the config values are conflicting.